packages/browser/__tests__/ClientAuthentication.spec.ts

```diff
@@ -318,4 +402,243 @@ describe("ClientAuthentication", () => {
+            isLoggedIn: "true",
+            issuer: "https://some.issuer",
+          },
+        }),
+        mockStorage({
+          [`${USER_SESSION_PREFIX}:${sessionId}`]: {
+            idToken: "some.id.token",
+          },
+        })
+      );
+      const clientAuthn = getClientAuthentication({
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the issuer does not have a JWKS", async () => {
+      const sessionId = "mySession";
+      // @ts-ignore
+      window.localStorage = new LocalStorageMock({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      const mockedIssuerConfig = mockIssuerConfigFetcher({} as IIssuerConfig);
+
+      const mockedStorage = await mockSessionStorage(sessionId);
+
+      const clientAuthn = getClientAuthentication({
+        issuerConfigFetcher: mockedIssuerConfig,
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the issuer's JWKS isn't available", async () => {
+      const sessionId = "mySession";
+      // @ts-ignore
+      window.localStorage = new LocalStorageMock({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      const mockedIssuerConfig = mockIssuerConfigFetcher({
+        jwksUri: "https://some.issuer/jwks",
+      } as IIssuerConfig);
+      const mockedStorage = await mockSessionStorage(sessionId);
+      mockFetch(new Response("Not a valid JWKS"));
+
+      const clientAuthn = getClientAuthentication({
+        issuerConfigFetcher: mockedIssuerConfig,
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the current issuer doesn't match the ID token's", async () => {
+      const sessionId = "mySession";
+      // @ts-ignore
+      window.localStorage = new LocalStorageMock({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      const mockedIssuerConfig = mockIssuerConfigFetcher({
+        jwksUri: "https://some.issuer/jwks",
+      } as IIssuerConfig);
+      const mockedStorage = await mockSessionStorage(
+        sessionId,
+        mockIdTokenPayload(
+          "https://my.pod/profile#me",
+          "https://some-other.issuer",
+          "https://some.app/registration"
+        )
+      );
+      mockFetch(new Response(JSON.stringify({ keys: [mockJwk()] })));
+
+      const clientAuthn = getClientAuthentication({
+        issuerConfigFetcher: mockedIssuerConfig,
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the current client ID doesn't match the ID token audience", async () => {
```

Comment:
It's not entirely clear to me which two values are the ones that shouldn't match for this test - is it `https://some-other.app/registratoin` and `https://some.issuer/jwks`? Maybe some clarifying comment could help for future readers.

Reply:
I think part of the issue is that some values come from the defaults st at the top of the file, which isn't really DAMP, and really isn't clear when reading the test. I think 26a099b4adc9a94574af9c7c6ad816c29cbb0ff5 will improve that.

Reply:
Yep clearer, thanks!